### ISSUE TITLE
Creating an f32 texture fails where an u8 one works

### ISSUE THREAD

**Issue by pedrocr** [glium/glium#1566]
I have a Vec<f32> with width * height * 3 elements (interleaved RGB). If I create a texture by first converting it to u8 it works:

```rust
let dims = (imgbuf.width as u32, imgbuf.height as u32);
let mut img = vec![0 as u8; imgbuf.data.len()];
for (o, i) in img.chunks_mut(1).zip(imgbuf.data.chunks(1)) {
  o[0] = (i[0]*255.0).min(255.0).max(0.0) as u8;
}
let raw_image = glium::texture::RawImage2d::from_raw_rgb_reversed(img, dims);
let img = glium::texture::Texture2d::new(&display, raw_image).unwrap();
```

but if I just create the texture from a copy of the original array:

```rust
let dims = (imgbuf.width as u32, imgbuf.height as u32);
let img = (*imgbuf).data.clone();
let raw_image = glium::texture::RawImage2d::from_raw_rgb_reversed(img, dims);
let img = glium::texture::Texture2d::new(&display, raw_image).unwrap();
rawid = Some(image_map.insert(img));
```

I get a 'Texture data size mismatch' panic in src/texture/any.rs:147 from the Texture2d::new call.

Am I using the API wrong somehow? It seems the only thing that changes is if I'm passing in Vec < u8 > or Vec < f32 >.

**Comment by tomaka** [comment]
Glium supposes that each element in the Vec in a single pixel.
If you want to change this behavior you have the possibility to manually implement the `Texture2dDataSource` trait.

**Comment by pedrocr** [comment]
from_raw_rgb(a)_*() already takes care of the interleaved stuff in the u8 case, why not in the f32 case?

**Comment by pedrocr** [comment]
Fixed by #1568
